apps/osim/workflows/default.yml

```diff
@@ -4,31 +4,41 @@ description: default vulnerability analysis
 priority: 0
 conditions: []
 states:
-  - name: DRAFT
+  - name: NEW
+    description: >
+      A new ticket arrived from incoming queue without any validation
+      or analysis performed yet
     requirements: []
 
-  - name: ANALYSIS
+  - name: TRIAGE
+    description: >
+      Queue captain has taken a ticket off the incoming queue and will
+      evalute if it should be rejected or qualify for further steps in
+      the workflow
     requirements:
-      - has affects
-      - has description
-      - has source
-      - has title
+      - state equals TRIAGE # check for manual change
```

Comment:
I previously had not the understanding that QC would be assigned and then unassigned, I will change that and it will became a problem since as you mentioned it shouldn't impact past states, one way I can immediately fix this would be having a higher priority workflow that starts manually (like in rejection), I can even put in the API a validation to unset assignment when entering in pre-secondary assessment, what do you think? will this be too ugly for the workflow files? I have submitted what I think can solve this, please take a look if it makes it better or worse

Reply:
Adding a special workflow to solve it or doing some hack for specific states does not seem to me as a good solutions. I was thinking about how to address this by the workflow YAML definition. It complicates it of course. The following solutions came to my mind.

First is probably more dirty. It would be introducing **or** condition. Something like `has owner or has team` which would be in the triage status. Then having `has team` in the pre-secondary and `has owner` in secondary. That would be adding the conditions in a way that removing the owner while at the same time adding the team between triage and pre-secondary would not violate it and then it would correctly accumulate. It may however be confusing for the reader and is not well aligned with the definition where in the triage stage there is really no team.

Second is probably cleaner but maybe more extending the definition. It would be adding some flag/mark/something that this requirement **is** or **is not** cumulative meaning that it contributes (or not) to the next states. I am thinking how to make this human readable. Maybe something like adding a requirement rule prefix like **for now has owner** which would tell the cumulative check to skip this requirement for the future state checks.

WDYT?